File: Artist/app/src/androidTest/java/com/example/artist/RecyclerViewItemsCountAssertion.java
package com.example.artist;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;
import androidx.test.espresso.NoMatchingViewException;
import androidx.test.espresso.ViewAssertion;

import org.hamcrest.Matcher;
import org.junit.Assert;

import static org.hamcrest.Matchers.is;


public class RecyclerViewItemsCountAssertion implements ViewAssertion {
    private Matcher<Integer> mMatcher;

    public RecyclerViewItemsCountAssertion(int expectedCount) {
        this.mMatcher = is(expectedCount);
    }

    @Override
    public void check(View view, NoMatchingViewException noViewFoundException) {
        if (noViewFoundException != null) {
            throw noViewFoundException;
        }

        RecyclerView recyclerView = (RecyclerView) view;
        RecyclerView.Adapter adapter = recyclerView.getAdapter();
        assert adapter != null;
        Assert.assertTrue(mMatcher.matches(adapter.getItemCount()));
    }
}

File: Artist/app/src/androidTest/java/com/example/artist/RecyclerViewItemsCountAssertion.kt
package com.example.artist

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.test.espresso.NoMatchingViewException
import androidx.test.espresso.ViewAssertion
import org.hamcrest.Matcher
import org.hamcrest.Matchers
import org.junit.Assert

class RecyclerViewItemsCountAssertion(expectedCount: Int) : ViewAssertion {
    private val mMatcher: Matcher<Int>
    override fun check(view: View, noViewFoundException: NoMatchingViewException) {
        if (noViewFoundException != null) {
            throw noViewFoundException
        }
        val recyclerView = view as RecyclerView
        val adapter = recyclerView.adapter!!
        Assert.assertTrue(mMatcher.matches(adapter.itemCount))
    }

    init {
        mMatcher = Matchers.`is`(expectedCount)
    }
}